Automattic/wp-calypso
# Datacenter Picker: Allow user to select datacenter when going Atomic

See https://github.com/Automattic/dotcom-forge/issues/1237

Some design conversation in p1669037882422489-slack-C041RHH38NQ
And more conversation in p1669156646360569-slack-C7YPW6K40
And a design request in p9Jlb4-5Xy-p2

## Proposed Changes

When a user takes their site Atomic from 'Hosting Configuration', they'll see a message about their site being automatically placed in the best datacenter:

<img width="842" alt="image" src="https://user-images.githubusercontent.com/36432/203596511-d744aef0-e815-4a46-826e-5955390bdb28.png">

If they click on 'Choose a datacenter instead', they're presented UI to pick their datacenter:

<img width="833" alt="image" src="https://user-images.githubusercontent.com/36432/203626960-23acbe0f-90fb-4482-85c5-847d980dee46.png">

If they don't have any eligibility warnings, they'll see this:

<img width="821" alt="image" src="https://user-images.githubusercontent.com/36432/203596908-2654f3bb-8267-4ca8-8224-5ec0dac732ce.png">

They won't be presented with the datapicker UI if they go Atomic outside of 'Hosting Configuration':

<img width="1171" alt="image" src="https://user-images.githubusercontent.com/36432/203597358-6fa73ba2-d2ea-489b-8f30-f1f730f0659e.png">

## Testing Instructions

1. First, sandbox `public-api.wordpress.com` and sandbox D93223-code.
2. Create a new WordPress.com site with a business plan.
3. Attempt to install a free plugin and verify the `<EligibilityWarnings>` component displays without the Datacenter Picker UI.
4. Navigate to 'Hosting Configuration' and click 'Activate'.
5. Verify the `<EligibilityWarnings>` component displays with the Datacenter Picker UI.
6. Verify the Datacenter Picker UI is usable, accessible, and looks good in the browsers we support.
7. If you want to recreate the 'no eligibility warnings' state, map a domain to your site and then go back to 'Hosting Configuration' -> 'Activate'.
8. Pick a custom datacenter, click 'Continue', and verify the site is created in the expected datacenter.
